classify_genes.R

```diff
@@ -108,6 +122,7 @@ classification = data.frame(gene_name = gene_names,
                             core = rep(0, length(gene_names)),
                             inter = rep(0, length(gene_names)),
                             rare = rep(0, length(gene_names)),
+                            singleton = rep(0, length(gene_names)),
```

Comment:
hmm I'm kind of confused because on the one hand you remove the singletons from the complete_presence_absence dataframe but then they do end up in this analysis? I think it should be kept as before, with filtering out all singleton genes from the curr_freqs df: 
```
curr_freqs = curr_freqs[!rownames(curr_freqs) %in% singleton_genes,]
```
Then remove the if-else block below and keep it as it was before. This slightly over-complicates the output and analysis for my liking, and I'd rather keep it that if singletons are removed they truly are removed from downstream analyses

Reply:
so they're included here because when not removed, I was still separating and reporting singletons throughout as an additional general classification (including in the graphs etc). But happy to amend and just give the option to remove them completely, or leave them in as part of the rare, lineage-specific set? (really sorry for extended delay in replying btw!)

Reply:
I guess it's ok to report them as separate class of genes, but then I would rename from "remove_singletons" to "separate_singletons", let me know when you have completed working on this and I will clone the branch to see how this works out in my dataset